lang/java/avro/src/test/java/org/apache/avro/TestSchemaCompatibility.java

```diff
@@ -515,4 +517,18 @@ public void testReaderWriterDecodingCompatibility() throws Exception {
           expectedDecodedDatum, decodedDatum);
     }
   }
+
+  @Test
+  public void testBytesDecimalWithSameScaleAndPrecision() {
```

Comment:
How about a negative test in here as well?

Reply:
Negative tests are located in different files: TestBytesSchemaCompatibilityDecimalMismatch and TestFixedSchemaCompatibilityDecimalMismatch.